ipa-core/src/protocol/ipa_prf/quicksort.rs

```diff
@@ -219,7 +230,7 @@ where
             list.swap(i - 1, pivot_index);
 
             // mark which ranges need to be sorted in the next pass
-            if i > pivot_index + 1 {
+            if i > pivot_index + 2 {
```

Comment:
`i` is initialized to `pivot_index + 1`. When we get here...

* if `i == pivot_index + 1`, then we didn't move any elements before the pivot.
* if `i == pivot_index + 2`, then we moved one element before the pivot, still no need to sort the before-pivot range n the next pass.

The other clue that something was off here is that one side of the if had +1 / -1, and the other had +1 / 0. +2 / -1 and +1 / 0 makes more sense.